### ISSUE TITLE
Improved SWAP flow

### ISSUE THREAD

**Issue by kilrau** [ExchangeUnion/xud#407]
My bad. I would like to propose to change the protocol for the taker to start the swap (as in my summary above) for the sake of saving the SWAP_REQUEST message. Also as @sangaman pointed out we can save all `DEAL` messages by sharing LND & Raiden PUBKEYs on handshake. I created an issue for this here:

Additionally, we'll put the maker in control via a `cancelSwap` which it can send after the taker did initiate the swap. Please also your opinion @sangaman 

0. LND & Raiden PUBKEYs get exchanged on handshake
1. Taker finds a match
2. Taker initiates the swap
once the swap is done Maker sends to Taker SWAP_RESPONSE which includes the r_preimage

**Comment by kilrau** [comment]
> What happens exactly in step 2

added details on that: hash, orderID, quantity in payload. That should work then!?

> I think in the current flow, we could do without the last 2 packets. 

As discussed in our call, I agree on this. So let's define this issue as a research task for offer if it's feasible to a) pack all this into the payload and b) how substantial the changes to lnd for that would be.

Elimination of the 2 packets in the next iteration: https://github.com/ExchangeUnion/xud/issues/421

**Comment by offerm** [comment]
See comment from #421 
```

@kilrau @sangaman hold horses on this. We are not done yet with the swap. Remember that we did the happy flow.

We still need to deal with expiry calculation of the HTLCs. For this we may need to share some information in the P2P packets (like the route and how many hopes are in route). This may also impact the decision if the maker should start the swap or the taker.

```

**Comment by kilrau** [comment]
New idea tbd: https://hackmd.io/s/SJ2qZUzdX#Swap-protocol-V3

**Comment by offerm** [comment]
@kilrau @sangaman is this still relevant?

**Comment by sangaman** [comment]
I don't think so. I'm pretty satisfied with how things work currently as far as the swap protocol.
